src/test/EnumerableSet.t.sol

```diff
@@ -0,0 +1,260 @@
+
+import {IFactory} from "alchemist/contracts/factory/IFactory.sol";
+
+import {
+    IUniversalVault,
+    Crucible
+} from "alchemist/contracts/crucible/Crucible.sol";
+import {CrucibleFactory} from "alchemist/contracts/crucible/CrucibleFactory.sol";
+import {ERC721Holder} from"@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import {MockERC20} from "../contracts/mocks/MockERC20.sol";
+
+import {EnumerableSet} from "../contracts/libraries/EnumerableSet.sol";
+import "forge-std/src/console2.sol";
+
+
+contract EnumerableSetTest is DSTest {
+    using EnumerableSet for EnumerableSet.TemplateDataSet;
+
+    /// @notice set of template data
+    EnumerableSet.TemplateDataSet internal templates; 
+
+    EnumerableSet.TemplateData internal DEFAULT;
+
+    Hevm cheats;
+
+    mapping(bytes32 => uint256) public gasBefore;
+    mapping(bytes32 => uint256) public measures;
+
+    function assertEq(bool a, bool b) internal {    
+        assertTrue(a == b);
+    }
+
+    function setUp() public {
+        cheats = Hevm(HEVM_ADDRESS);
+
+        DEFAULT = EnumerableSet.TemplateData({
+            template: address(1337),
+            disabled: false,
+            name: 'template'
+        });
+    }
+    
+
+    function test_at_index() public {
+        templates.add(DEFAULT);
+        EnumerableSet.TemplateData memory value = templates.at(0);
+
+        assertEq(value.template, address(1337));
+        assertEq(value.disabled, false);
+        assertEq(value.name, 'template');
+    }
+
+    function test_at_template() public {
+        templates.add(DEFAULT);
+
+        EnumerableSet.TemplateData memory value = templates.at(address(1337));
+
+        assertEq(value.template, address(1337));
+        assertEq(value.disabled, false);
+        assertEq(value.name, 'template');
+    }
+
+    // function 
+
+    function test_contains(uint256 length) public {
+        cheats.assume(length < 50);
+
+        for (uint i = 0; i < length; i++) {
+            assertTrue(templates.add(EnumerableSet.TemplateData({
+                template: address(uint160(1337+i)),
+                name: string(abi.encode('template', i)),
+                disabled: false
+            })));
+            assertTrue(templates.contains(address(uint160(1337+i))));
+        }
+
+        
+        assertEq(templates.length(), length);
+        assertTrue(!templates.contains(address(uint160(1337+length))));
+        
+    }
+
+
+    // function test_contains(uint256 length) public {
+    //     cheats.assume(length < 50);
+
+    //     for (uint i = 0; i < length; i++) {
+    //         bool v = templates.add(EnumerableSet.TemplateData({
+    //             template: address(uint160(1337+i)),
+    //             name: string(abi.encode('template', i)),
+    //             disabled: false
+    //         }));
+    //         assertTrue(v);
+    //     }
+
+    //     assertEq(templates.length(), length);
+    // }
```

Comment:
This was commented for a reason. We should know why.

Reply:
I wonder who we could ask about it:

![image](https://user-images.githubusercontent.com/111788132/193078547-e596586e-7b94-4813-8d7a-0580b8960ef7.png)

Reply:
the next iteration of this testcase is implemented right above it, I'll remove this